var/spack/packages/hwloc/package.py

```diff
@@ -15,15 +15,17 @@ class Hwloc(Package):
     homepage = "http://www.open-mpi.org/projects/hwloc/"
     url      = "http://www.open-mpi.org/software/hwloc/v1.9/downloads/hwloc-1.9.tar.gz"
 
+    version('1.11.2', '486169cbe111cdea57be12638828ebbf',
+            url='http://www.open-mpi.org/software/hwloc/v1.11/downloads/hwloc-1.11.2.tar.bz2')
     version('1.11.1', '002742efd3a8431f98d6315365a2b543',
             url='http://www.open-mpi.org/software/hwloc/v1.11/downloads/hwloc-1.11.1.tar.bz2')
     version('1.9', '1f9f9155682fe8946a97c08896109508')
 
-    depends_on('libpciaccess')
+    # libpciaccess is not supported on OS X
+    depends_on('libpciaccess', when='=linux')
```

Comment:
Let me see what i can add -- one reason this is complicated is that the `spec` syntax doesn't really have a way to say "not" for architectures right now -- `Specs` are really match objects and they only match in the positive.

A simple way to do this would be something like:
```python
depends_on('libpciaccess', when='!=darwin-x86_64')
```
that would be much clearer than the above proposals.  I would need to think through how the spec syntax should be modified to support this, but it is something we need.

The other option would be to just add "when_not", which might be easier short-term.  That would look like this:
```python
depends_on('libpciaccess', when_not='=darwin-x86_64')
```
You could test either of these in `install` by just writing:
```python
    if 'libpciaccess' in spec:
        ...
```

Would either of those work for you?